### ISSUE TITLE
Enhancement: [naming-convention] Allow selecting only const with <const>literal

### ISSUE THREAD

**Issue by nowaysgit** [typescript-eslint/typescript-eslint#10530]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/naming-convention

### Description

```
{
  "rules": {
    "@typescript-eslint/naming-convention": ['error',
    {
        selector: 'variable',
        format: ['UPPER_CASE'],
        **modifiers: ['constantLiteral'],** // need it
    },
  }]
}
```

### Fail

```TypeScript
const ipAddress = <const>'100.100.100.10';
```

### Pass

```TypeScript
const IP_ADDRESS = <const>'100.100.100.10';
```

**Comment by JoshuaKGoldberg** [comment]
From the rule's docs that you linked:

> This rule is _feature frozen_: it will no longer receive new features such as new options. It still will accept bug and documentation fixes for its existing area of features and to support new TypeScript versions.
>
> Stylistic rules that enforce naming and/or sorting conventions tend to grow incomprehensibly complex as increasingly obscure features are requested. This rule has reached the limit of what is reasonable for the typescript-eslint project to maintain. See [eslint-plugin: Feature freeze naming and sorting stylistic rules](https://github.com/typescript-eslint/typescript-eslint/issues/8792) for more information.

Thanks for filing, but this is out of scope for what we have the maintenance bandwidth to maintain here. If you really want this, I'd suggest either writing your own rule ([Developers > Building Custom Rules](https://typescript-eslint.io/developers/custom-rules)) or forking the naming-convention rule.
